### DISCUSSION TITLE
compilation failed

### DISCUSSION THREAD

**Discussion by zeynabhoshyar** [chatwoot/chatwoot#7603] [Self hosted] [answered]
I install chatwoot by self hosted guide, I try to change some css in chatwoot widget code in app/javascript/widget, like change some translation text, add font, direction of widget. I use this commands: 
sudo -i -u chatwoot 
cd chatwoot 
rake assets:precompile RAILS_ENV=production
but compilation faild, and I dont know why? please help me to fix it.

**compile result:**
--------------------------------
yarn install v1.22.19
[1/5] Validating package.json...
[2/5] Resolving packages...
success Already up-to-date.
$ husky install
husky - Git hooks installed
Done in 1.54s.
Compiling...
Compilation failed:

warn - You have enabled experimental features: darkModeVariant
warn - Experimental features are not covered by semver, may introduce breaking changes, and can change at any time.

warn - The `conservative` purge mode will be removed in Tailwind 2.0.
warn - Please switch to the new `layers` mode instead.
DEPRECATION WARNING: Using / for division outside of calc() is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($num, $num * 0 + 1) or calc($num / ($num * 0 + 1))

More info and automated migrator: https://sass-lang.com/d/slash-div

   ╷
17 │   @return $num / ($num * 0 + 1);
   │           ^^^^^^^^^^^^^^^^^^^^^
   ╵
    node_modules/foundation-sites/scss/util/_unit.scss 17:11             strip-unit()
    node_modules/foundation-sites/scss/util/_unit.scss 89:13             -zf-to-rem()
    node_modules/foundation-sites/scss/util/_unit.scss 47:13             rem-calc()
    node_modules/foundation-sites/scss/util/_unit.scss 61:13             -zf-bp-to-em()
    node_modules/foundation-sites/scss/util/_breakpoint.scss 78:8        breakpoint()
    node_modules/foundation-sites/scss/util/_breakpoint.scss 329:38      @import
    node_modules/foundation-sites/scss/util/_util.scss 12:9              @import
    app/javascript/dashboard/assets/scss/_foundation-settings.scss 44:9  @import
    app/javascript/dashboard/assets/scss/_woot.scss 14:9                 @import
    app/javascript/dashboard/assets/scss/app.scss 1:9                    root stylesheet

DEPRECATION WARNING: Using / for division outside of calc() is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div(strip-unit($value), strip-unit($base)) or calc(strip-unit($value) / strip-unit($base))

More info and automated migrator: https://sass-lang.com/d/slash-div
....

**Comment by ahmed233244** [comment]
did you solve it?

**Comment by zeynabhoshyar** [comment]
yes, I solve it.

**Reply by jdabeau** [reply]
how did you solve it?

**Reply by zeynabhoshyar** [reply, marked as answer]
My server resources were not enough for processing, I increased it.
